P2P sanity test failing

Using:
  Android 1.4-26
  Mac 1.4-29

Failing test - http://uberjenkins.sc.couchbase.com/job/android-listener-client-client/19/testReport/testsuites.listener.shared.client_client/test_peer_2_peer_sanity/test_peer_2_peer_sanity/

```
V/Listener( 5373): CountDownLatch.await() END
D/dalvikvm( 5373): GC_FOR_ALLOC freed 435K, 11% free 7966K/8903K, paused 3ms, total 3ms
W/ChangeTracker( 5373): ChangeTracker{http://10.17.1.40:59850/ls_db2, OneShot, @535aed34}: Exception in change tracker
W/ChangeTracker( 5373): java.net.SocketTimeoutException: timeout
W/ChangeTracker( 5373): 	at okio.Okio$3.newTimeoutException(Okio.java:212)
W/ChangeTracker( 5373): 	at okio.AsyncTimeout.exit(AsyncTimeout.java:288)
W/ChangeTracker( 5373): 	at okio.AsyncTimeout$2.read(AsyncTimeout.java:242)
W/ChangeTracker( 5373): 	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:325)
W/ChangeTracker( 5373): 	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:314)
W/ChangeTracker( 5373): 	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:210)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.Http1xStream.readResponse(Http1xStream.java:184)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:125)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:775)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.HttpEngine.access$200(HttpEngine.java:86)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.HttpEngine$NetworkInterceptorChain.proceed(HttpEngine.java:760)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.HttpEngine.readResponse(HttpEngine.java:613)
W/ChangeTracker( 5373): 	at okhttp3.RealCall.getResponse(RealCall.java:244)
W/ChangeTracker( 5373): 	at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:201)
W/ChangeTracker( 5373): 	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:163)
W/ChangeTracker( 5373): 	at okhttp3.RealCall.execute(RealCall.java:57)
W/ChangeTracker( 5373): 	at com.couchbase.lite.replicator.ChangeTracker.runLoop(ChangeTracker.java:336)
W/ChangeTracker( 5373): 	at com.couchbase.lite.replicator.ChangeTracker.run(ChangeTracker.java:246)
W/ChangeTracker( 5373): 	at java.lang.Thread.run(Thread.java:856)
W/ChangeTracker( 5373): Caused by: java.net.SocketException: Socket closed
W/ChangeTracker( 5373): 	at libcore.io.Posix.recvfromBytes(Native Method)
W/ChangeTracker( 5373): 	at libcore.io.Posix.recvfrom(Posix.java:131)
W/ChangeTracker( 5373): 	at libcore.io.BlockGuardOs.recvfrom(BlockGuardOs.java:164)
W/ChangeTracker( 5373): 	at libcore.io.IoBridge.recvfrom(IoBridge.java:513)
W/ChangeTracker( 5373): 	at java.net.PlainSocketImpl.read(PlainSocketImpl.java:488)
W/ChangeTracker( 5373): 	at java.net.PlainSocketImpl.access$000(PlainSocketImpl.java:46)
W/ChangeTracker( 5373): 	at java.net.PlainSocketImpl$PlainSocketInputStream.read(PlainSocketImpl.java:240)
W/ChangeTracker( 5373): 	at okio.Okio$2.read(Okio.java:140)
W/ChangeTracker( 5373): 	at okio.AsyncTimeout$2.read(AsyncTimeout.java:238)
W/ChangeTracker( 5373): 	... 16 more
V/ChangeTracker( 5373): ChangeTracker{http://10.17.1.40:59850/ls_db2, OneShot, @535aed34}: Making request to POST [http://10.17.1.40:59850/ls_db2/_changes?feed=longpoll&heartbeat=30000&style=all_docs&active_only=true]
D/Listener( 5373): authHeader is null
V/Listener( 5373): Not enforcing basic auth -- allowedCredentials null or empty
V/Listener( 5373): remoteURL() addr=10.17.1.40
V/Listener( 5373): RouterCallbackBlock.onResponseReady() START
V/Listener( 5373): RouterCallbackBlock.onResponseReady() END
V/Listener( 5373): CountDownLatch.await() START
V/Listener( 5373): CountDownLatch.await() END
D/dalvikvm( 5373): GC_CONCURRENT freed 324K, 10% free 8030K/8903K, paused 11ms+0ms, total 15ms
W/ChangeTracker( 5373): ChangeTracker{http://10.17.1.40:59850/ls_db2, OneShot, @535aed34}: Exception in change tracker
W/ChangeTracker( 5373): java.net.SocketTimeoutException
W/ChangeTracker( 5373): 	at java.net.PlainSocketImpl.read(PlainSocketImpl.java:491)
W/ChangeTracker( 5373): 	at java.net.PlainSocketImpl.access$000(PlainSocketImpl.java:46)
W/ChangeTracker( 5373): 	at java.net.PlainSocketImpl$PlainSocketInputStream.read(PlainSocketImpl.java:240)
W/ChangeTracker( 5373): 	at okio.Okio$2.read(Okio.java:140)
W/ChangeTracker( 5373): 	at okio.AsyncTimeout$2.read(AsyncTimeout.java:238)
W/ChangeTracker( 5373): 	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:325)
W/ChangeTracker( 5373): 	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:314)
W/ChangeTracker( 5373): 	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:210)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.Http1xStream.readResponse(Http1xStream.java:184)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:125)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:775)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.HttpEngine.access$200(HttpEngine.java:86)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.HttpEngine$NetworkInterceptorChain.proceed(HttpEngine.java:760)
W/ChangeTracker( 5373): 	at okhttp3.internal.http.HttpEngine.readResponse(HttpEngine.java:613)
W/ChangeTracker( 5373): 	at okhttp3.RealCall.getResponse(RealCall.java:244)
W/ChangeTracker( 5373): 	at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:201)
W/ChangeTracker( 5373): 	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:163)
W/ChangeTracker( 5373): 	at okhttp3.RealCall.execute(RealCall.java:57)
W/ChangeTracker( 5373): 	at com.couchbase.lite.replicator.ChangeTracker.runLoop(ChangeTracker.java:336)
W/ChangeTracker( 5373): 	at com.couchbase.lite.replicator.ChangeTracker.run(ChangeTracker.java:246)
W/ChangeTracker( 5373): 	at java.lang.Thread.run(Thread.java:856)
V/ChangeTracker( 5373): ChangeTracker{http://10.17.1.40:59850/ls_db2, OneShot, @535aed34}: Making request to POST [http://10.17.1.40:59850/ls_db2/_changes?feed=longpoll&heartbeat=30000&style=all_docs&active_only=true]
```

@sethrosetter:
The test fails because `macosx`'s  pull replication status never becomes `IDLE`. I assume this is CBL iOS issue. There is possibility that CBL Android returns the unexpected return value to CBL iOS.

Error message from pytest:
```
        except _ProxyError as e:
            raise ProxyError(e)
    
        except (_SSLError, _HTTPError) as e:
            if isinstance(e, _SSLError):
                raise SSLError(e, request=request)
            elif isinstance(e, ReadTimeoutError):
>               raise ReadTimeout(e, request=request)
E               ReadTimeout: HTTPConnectionPool(host='10.17.1.60', port=51000): Read timed out. (read timeout=None)

venv/lib/python2.7/site-packages/requests/adapters.py:449: ReadTimeout

Tearing down test
```

android - http://192.168.57.101:5000/_active_tasks
```json
[
{
progress: 0,
continuous: true,
source: "http://10.17.1.60:51000/ls_db2",
status: "Idle",
task: "repl002",
change_count: 0,
target: "ls_db1",
completed_change_count: 0,
type: "Replication"
},
{
progress: 0,
continuous: true,
source: "ls_db1",
status: "Idle",
task: "repl001",
change_count: 0,
target: "http://10.17.1.60:51000/ls_db2",
completed_change_count: 0,
type: "Replication"
}
]
```
macosx - http://10.17.1.60:51000/_active_tasks
```json
[
{
continuous: true,
status: "Idle",
task: "repl001",
type: "Replication",
source: "ls_db2",
target: "http://192.168.57.101:5000/ls_db1"
},
{
continuous: true,
status: "Processed 0 / 0 changes",
task: "repl002",
type: "Replication",
source: "http://192.168.57.101:5000/ls_db1",
target: "ls_db2"
}
]
```

--- a/src/main/java/com/couchbase/lite/Manager.java
+++ b/src/main/java/com/couchbase/lite/Manager.java
@@ -24,6 +24,7 @@ import com.couchbase.lite.support.Version;
 import com.couchbase.lite.util.Log;
 import com.couchbase.lite.util.StreamUtils;
 import com.couchbase.lite.util.Utils;
+import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 import java.io.File;
@@ -69,7 +70,10 @@ public final class Manager {
     public static final String FORESTDB_STORAGE = "ForestDB";
 
     // NOTE: Jackson is thread-safe http://wiki.fasterxml.com/JacksonFAQThreadSafety
-    private static final ObjectMapper mapper = new ObjectMapper();
+    // NOTE: Jackson automatically close input resource after parsing the conent.
+    //       Need to disable it. https://github.com/FasterXML/jackson-databind/issues/697
+    private static final ObjectMapper mapper = new ObjectMapper()
+            .disable(JsonParser.Feature.AUTO_CLOSE_SOURCE);
 
     private ManagerOptions options;
     private File directoryFile;

--- a/src/main/java/com/couchbase/lite/router/Router.java
+++ b/src/main/java/com/couchbase/lite/router/Router.java
@@ -244,6 +244,7 @@ public class Router implements Database.ChangeListener, Database.DatabaseListene
             throw new CouchbaseLiteException(Status.NOT_ACCEPTABLE);
 
         // parse body text
+        // NOTE: contentStream should not be close immediately. It will close Socket connection.
         InputStream contentStream = connection.getRequestInputStream();
         try {
             return Manager.getObjectMapper().readValue(contentStream, Map.class);

